include/nuttx/net/netconfig.h

```diff
@@ -104,52 +104,48 @@
 
 #define NET_SOCK_PROTOCOL  0
 
-/* SOCK_DGRAM is the preferred socket type to use when we just want a
- * socket for performing driver ioctls.  However, we can't use SOCK_DRAM
- * if UDP is disabled.
- *
- * Pick a socket type (and perhaps protocol) compatible with the currently
- * selected address family.
+/* SOCK_CTRL is the preferred socket type to use when we just want a
+ * socket for performing driver ioctls.
  */
 
 #if NET_SOCK_FAMILY == AF_INET
 #  if defined(CONFIG_NET_UDP)
-#    define NET_SOCK_TYPE SOCK_DGRAM
+#    define NET_SOCK_TYPE SOCK_CTRL
```

Comment:
let's alwasy define
NET_SOCK_TYPE SOCK_CTRL and simplify the logic to define NET_SOCK_PROTOCOL

Reply:
Sure, I'll do it soon.

Reply:
@xiaoxiang781216 san
I updated a changes. Please review again.